Apps/CocoaPods-FCM/src/App.swift

```diff
@@ -6,16 +6,34 @@ struct MainApp: App {
 
     @StateObject var userManager: UserManager = .init()
 
+    @State private var openedDeepLinkUrl: URL?
+
     var body: some Scene {
         WindowGroup {
-            if userManager.isUserLoggedIn {
-                DashboardView()
-                    .environmentObject(userManager)
-                    .accentColor(Color("AccentColor")) // sets Color.accentColor for all children
-            } else {
-                LoginView()
-                    .environmentObject(userManager)
-                    .accentColor(Color("AccentColor")) // sets Color.accentColor for all children
+            HStack {
+                if userManager.isUserLoggedIn {
+                    DashboardView()
+                        .environmentObject(userManager)
+                        .accentColor(Color("AccentColor")) // sets Color.accentColor for all children
+                } else {
+                    LoginView()
+                        .environmentObject(userManager)
+                        .accentColor(Color("AccentColor")) // sets Color.accentColor for all children
+                }
+            }.onOpenURL { deepLink in // This function is how to implement deep links in a SwiftUIU app.
+                // This app opens deep links using Universal Links and app scheme deep links.
+                //
+                // Universal Links: Any URL that begins with `https://ciosample.page.link`...
+                // App scheme: Any URL that begins with `cocoapods-fcm://`...
+                //
+                // ...will open the app and display the deep link in a pop-up.
+                openedDeepLinkUrl = deepLink
+            }.alert(isPresented: Binding<URL>.notNil(openedDeepLinkUrl)) {
+                Alert(
+                    title: Text("Deep link opened!"),
+                    message: Text(openedDeepLinkUrl!.absoluteString),
```

Comment:
Can we possibly redirect to a secreen? as a sample for users?

Reply:
+1 to that.
I have added a separate screen to my apps specifically for deep links and universal links, which was not included in the original design. I believe it adds value to the user experience and QA as well.

Reply:
I can see how that could be useful. Since this isn't part of the original ticket or design, I am going to skip that for this pull request.